crates/wash-lib/src/cli/inspect.rs

```diff
@@ -92,14 +102,33 @@ pub async fn handle_command(
         let caps = get_caps(command.clone(), &buf).await?;
         let token =
             caps.ok_or_else(|| anyhow!("No capabilities discovered in : {}", module_name))?;
-        if jwt_only {
-            let out = CommandOutput::from_key_and_text("token", token.jwt);
-            Ok(out)
+        Ok(if command.wit {
+            let witty = wit_component::decode(&buf).expect("Failed to decode WIT");
+            let resolve = witty.resolve();
+            let main = witty.package();
+            // NOTE(brooksmtownsend): This is a reasonable way that I found to detect WASI Preview 1 Wasms.
+            // Happy to replace with any level of additional detail if that's available.
+            if resolve.worlds.len() == 1 {
```

Comment:
We could probably just use `wasmparser` check whether this is a module or a component

Reply:
Good point!

Reply:
@rvolosatovs decently refactored the logic in the inspect function to just try and parse the `buf` as a Wasm, and based on wasmparser's result + CLI flags branch from there. Let me know if anything is wrong there